src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.Unix.cs

```diff
@@ -33,9 +33,30 @@ internal static unsafe int ReadAtOffset(SafeFileHandle handle, Span<byte> buffer
                 // The Windows implementation uses ReadFile, which ignores the offset if the handle
                 // isn't seekable.  We do the same manually with PRead vs Read, in order to enable
                 // the function to be used by FileStream for all the same situations.
-                int result = handle.CanSeek ?
-                    Interop.Sys.PRead(handle, bufPtr, buffer.Length, fileOffset) :
-                    Interop.Sys.Read(handle, bufPtr, buffer.Length);
+                int result;
+                if (handle.SupportsRandomAccess)
+                {
+                    // Try pread for seekable files.
+                    result = Interop.Sys.PRead(handle, bufPtr, buffer.Length, fileOffset);
+                    if (result == -1)
+                    {
+                        // Some devices (such as /dev/tty) are reported as seekable by macOS but pread is unable to handle them and returns ENXIO.
+                        // Historically we were able to handle /dev/tty using read so we need to fallback to read for that case.
+                        Interop.ErrorInfo errorInfo = Interop.Sys.GetLastErrorInfo();
+                        AssertPReadPWriteKnownErrors(errorInfo.Error);
```

Comment:
This `Assert` doesn't bring much. Any error not handled specifically will be visible as an `IOException`.

Reply:
It helps to double-check that the exception is one already expected when someone writes a test with `Assert.Throws<IOException>(fs.Write())`.
IMO, `IOException` may be too ambiguous.

Reply:
The man page and POSIX docs list more valid errnos than the ones that `AssertPReadPWriteKnownErrors` accepts. This is overly strict and can cause a `Debug.Assert` when it is not desired.